src/BSeries.jl

```diff
@@ -1453,4 +1456,399 @@ include("latexify.jl")
     include("precompile.jl")
 end
 
+
+
+"""
+        is_energy_preserving(rka::RungeKuttaMethod,order)::Bool
+
+This function checks whether a Runge-Kutta method `rk` is 
+energy-preserving for a given `order`.
+"""
+function is_energy_preserving(rk::RungeKuttaMethod,order)
+#generate bseries 
+    series = bseries(rk, order)
+    #pass it to the main function 'is_energy_preserving'
+    is_energy_preserving(series)
+end
+
+"""
+    is_energy_preserving(map_series)::Bool
+This function checks whether the B-series `series` of a time integration
+method is energy-preserving for Hamiltonian systems - up to the
+[`order`](@ref) of the `series`.
+
+# References
+This code is based on the Theorem 2 of
+- Elena Celledoni, Robert I. McLachlan, Brynjulf Owren, and G. R. W. Quispel. 
+  "Energy-preserving integrators and the structure of B-series."
+  Foundations of Computational Mathematics 10 (2010): 673-693.
+  [DOI: 10.1007/s10208-010-9073-1](https://link.springer.com/article/10.1007/s10208-010-9073-1)
+ """
+function is_energy_preserving(map_series)
+    flow_series = modified_equation(map_series)
+    #check if the first element is map or flow
+    @assert flow_series[rootedtree([1])] == 1
+    #renormalize by multiplying by symmetry factor
+    renormalize_bseries!(flow_series)
+    #save all the coefficients in an array: it is easier
+    #to call an array full of fractions when working with the 
+    #functions 'renormalize_bseries!' and 'energy_preserving_trees_test' 
+    coefficients = collect(values(flow_series))
+    #save all the RootedTrees in another array: 
+    thetrees = collect(keys(flow_series))
+    #we need only the level sequence
+    # Create an empty vector to store the converted trees into arrays
+    trees = Vector{Vector{Int}}(undef, length(flow_series))
+    # Convert the trees and store them in the 'trees' vector
+    for i in 1:length(flow_series)
+        trees[i] = thetrees[i].level_sequence
+    end
+    energy_preserving_flag = true
+    max_length = length(trees[end])
+    l = length(trees)
+    #first, check if the energy_preserving condition is satisfied
+    #low order (provided that the computational cost at order < 5)
+    # is low. This is because the 'rank' function's behavior
+    # in the test function doesn't works by separate for low orders.
```

Comment:
I don't understand this comment. Could you please explain it to me?

Reply:
@Sondar74 edited this earlier today but I guess he hasn't pushed it.  @Sondar74 please push your changes so @ranocha can do a final review.

Reply:
Thanks!